httpd/handler.go

```diff
@@ -174,11 +178,94 @@ func (h *Handler) serveQuery(w http.ResponseWriter, r *http.Request, user *influ
 		return
 	}
 
-	// Execute query. One result will return for each statement.
-	results := h.server.ExecuteQuery(query, db, user)
+	// get the chunking settings
+	chunked := q.Get("chunked") == "true"
+	// even if we're not chunking, the engine will chunk at this size and then the handler will combine results
+	chunkSize := DefaultChunkSize
+	if chunked {
+		if cs, err := strconv.ParseInt(q.Get("chunk_size"), 10, 64); err == nil {
+			chunkSize = int(cs)
+		}
+	}
 
 	// Send results to client.
-	httpResults(w, results, pretty)
+	w.Header().Add("content-type", "application/json")
+	results, err := h.server.ExecuteQuery(query, db, user, chunkSize)
+	if err != nil {
+		if isAuthorizationError(err) {
+			w.WriteHeader(http.StatusUnauthorized)
+		} else {
+			w.WriteHeader(http.StatusInternalServerError)
+		}
+		return
+	}
+
+	// if we're not chunking, this will be the in memory buffer for all results before sending to client
+	res := influxdb.Results{Results: make([]*influxdb.Result, 0)}
+	statusWritten := false
+
+	// pull all results from the channel
+	for r := range results {
+		// write the status header based on the first result returned in the channel
+		if !statusWritten {
```

Comment:
If something goes wrong later on, how do we communicate to the client that we broke?  I think this is the correct approach here, but this is a similar scenario to the `dump` command, that once things go wrong, we may need to signal that we had a problem and couldn't continue.